azkaban-common/src/main/java/azkaban/event/EventData.java

```diff
@@ -0,0 +1,34 @@
+package azkaban.event;
+
+import azkaban.executor.Status;
+
+/**
+ * Carries a snapshot of the status data, suitable for asynchronous message passing.
+ */
+public class EventData {
+
+  private final Status status;
+  private final String nestedId;
+
+  public static EventData create(Status status) {
+    return create(status, null);
+  }
+
+  public static EventData create(Status status, String nestedId) {
```

Comment:
Yeah, indeed I only did that because `Event` uses that pattern. I now modified EventData to have just public constructors.